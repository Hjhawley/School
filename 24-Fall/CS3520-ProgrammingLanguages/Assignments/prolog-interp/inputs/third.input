third(cons(_1,cons(_2,cons(Y,_3))),Y).

third(cons(v1,cons(v2,cons(v3,cons(v4,cons(v5,nil))))),v3).
third(cons(v2,cons(v3,cons(v4,cons(v5,nil)))),v4).
third(cons(a,cons(b,cons(c,cons(d,nil)))),c).
third(cons(a,cons(b,cons(c,nil))),c).
third(cons(a,cons(b,cons(c,d))),c).
third(cons(v1,cons(v2,cons(v3,cons(v4,nil)))),v2).
third(cons(v1,cons(v2,cons(v3,cons(v4,nil)))),v4).
third(cons(a,cons(b,cons(c,cons(d,nil)))),b).
third(cons(a,cons(b,cons(c,cons(d,nil)))),d).
third(cons(a,cons(b,nil)),a).
third(notalist,a).
