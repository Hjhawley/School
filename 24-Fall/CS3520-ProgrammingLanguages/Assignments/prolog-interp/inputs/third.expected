
Please enter clauses (blank line to end the database):
tokens: [third ( cons ( _1 , cons ( _2 , cons ( Y , _3 ) ) ) , Y ) .]
clause: third([_1,_2,Y|_3],Y).

Please enter a query (blank line to quit):
tokens: [third ( cons ( v1 , cons ( v2 , cons ( v3 , cons ( v4 , cons ( v5 , nil ) ) ) ) ) , v3 ) .]
query: third([v1,v2,v3,v4,v5],v3).
SUCCESS: third([v1,v2,v3,v4,v5],v3)

Please enter a query (blank line to quit):
tokens: [third ( cons ( v2 , cons ( v3 , cons ( v4 , cons ( v5 , nil ) ) ) ) , v4 ) .]
query: third([v2,v3,v4,v5],v4).
SUCCESS: third([v2,v3,v4,v5],v4)

Please enter a query (blank line to quit):
tokens: [third ( cons ( a , cons ( b , cons ( c , cons ( d , nil ) ) ) ) , c ) .]
query: third([a,b,c,d],c).
SUCCESS: third([a,b,c,d],c)

Please enter a query (blank line to quit):
tokens: [third ( cons ( a , cons ( b , cons ( c , nil ) ) ) , c ) .]
query: third([a,b,c],c).
SUCCESS: third([a,b,c],c)

Please enter a query (blank line to quit):
tokens: [third ( cons ( a , cons ( b , cons ( c , d ) ) ) , c ) .]
query: third([a,b,c|d],c).
SUCCESS: third([a,b,c|d],c)

Please enter a query (blank line to quit):
tokens: [third ( cons ( v1 , cons ( v2 , cons ( v3 , cons ( v4 , nil ) ) ) ) , v2 ) .]
query: third([v1,v2,v3,v4],v2).

Please enter a query (blank line to quit):
tokens: [third ( cons ( v1 , cons ( v2 , cons ( v3 , cons ( v4 , nil ) ) ) ) , v4 ) .]
query: third([v1,v2,v3,v4],v4).

Please enter a query (blank line to quit):
tokens: [third ( cons ( a , cons ( b , cons ( c , cons ( d , nil ) ) ) ) , b ) .]
query: third([a,b,c,d],b).

Please enter a query (blank line to quit):
tokens: [third ( cons ( a , cons ( b , cons ( c , cons ( d , nil ) ) ) ) , d ) .]
query: third([a,b,c,d],d).

Please enter a query (blank line to quit):
tokens: [third ( cons ( a , cons ( b , nil ) ) , a ) .]
query: third([a,b],a).

Please enter a query (blank line to quit):
tokens: [third ( notalist , a ) .]
query: third(notalist,a).

Please enter a query (blank line to quit):
